smoketest.md

```diff
@@ -0,0 +1,102 @@
+```
+MONGOSHELL=$(find ./local_adf/ | grep mongo.exe | head -1) 
+chmod +x $MONGOSHELL
+$MONGOSHELL -u $ADF_TEST_LOCAL_USER --password $ADF_TEST_LOCAL_PWD --authenticationDatabase \
+            $ADF_TEST_LOCAL_AUTH_DB $ADF_TEST_LOCAL_HOST/admin \
+            --eval 'db.runCommand({sqlGenerateSchema: 1, 
+            sampleNamespaces: ["integration_test.complex_types", "supplies.sales"], setSchemas: true})'
+```
+#### Clear Power BI Cache
+It is a good idea to clear the data cache and data source settings to ensure the connector is being correctly tested.
+##### Clear Data Cache
+* Navigate to `File` -> `Options and settings` -> `Options` -> `Data Load`
+* Under `Data Cache Management Options` click on `Clear Cache`
+
+##### Clear Saved Data Source Settings
+* Navigate to `File` -> `Options and settings` -> `Data source settings`
+* For `Data sources in current file` and `Global permissions` choose `Clear Permissions`->`Clear All Permissions` 
+
+## Running Test
+### Navigation Table
+Test that the expected tables are shown in the navigation table and that data is loaded in the expected format.
+* `Get Data` -> `More...` -> `Database` -> `MongoDB Atlas SQL (Beta)`
+* Enter MongoDB URI: `mongodb://localhost/?ssl=false`
+* Enter Database: `integration_test`
+* Click `OK`
+* In the Navigator, expand the `integration_test` and `supplies` databases to show the underlying collections
+* Choose the `complex_types` collection and verify that the expected preview loads
+* Click `Transform Data`
+* Update the query to the following to show the table schema:
+```
+= Table.Schema(integration_test_Database{[Name="complex_types",Kind="Table"]}[Data])
+```
+* Confirm the `TypeName` and `NativeTypeName` columns have the expected values
+  * The `TypeName` should be `Text.Type` for the complex types 
+* Transform the `array` column to JSON
+* Expand the list to new rows
+* Click `Close and Apply`
+* In `Visualizations` choose `Clustered column chart`
+* In `Data` choose `double` and `integer`, verify the visualization is as expected
+
+### Query
+Test that data is loaded in the expected format when running a native query.
+* `Get Data` -> `More...` -> `Database` -> `MongoDB Atlas SQL (Beta)`
+* Enter MongoDB URI: `mongodb://localhost/?ssl=false`
+* Enter Database: `supplies`
+* Enter SQL Statement: `select * from sales`
+* Click `OK`
+* Verify that the expected preview loads
+* Click `Transform Data`
+* Confirm that the type is `Text` for all the columns with complex types
+* Transform the `customer` column to JSON
+* Expand all the values in the resulting `customer` records
+* Change types of `customer.age` and `customer.satisfaction` to Whole Number
+* Click `Close and Apply`
+* In `Visualizations` choose `Clustered column chart`
+* In `Data` for the Query choose `customer.age`, `customer.satisfaction`, and `_id`, verify the visualization is as expected
+
+### On-Premises Data Gateway
```

Comment:
Much of this section is from what @jchemburkar added to the Design doc when adding Data Gateway support.